graph/GuidelinesGraph.md

```diff
@@ -0,0 +1,456 @@
+  - [Design Approach](#design-approach)
+    - [Naming](#naming)
+    - [Uniform Resource Locators (URLs)](#uniform-resource-locators-urls)
+    - [Query Support](#query-support)
+    - [Resource Modeling Patterns](#resource-modeling-patterns)
+    - [Behavior Modeling](#behavior-modeling)
+    - [Error Handling](#error-handling)
+  - [API contract and non-backward compatible changes](#api-contract-and-non-backward-compatible-changes)
+    - [Versioning and Deprecation](#versioning-and-deprecation)
+  - [Recommended API Patterns](#recommended-api-patterns)
+  - [References](#references)
+
+## 
+
+#### History
+
+| Date        | Notes                       |
+|-------------|-----------------------------|
+| 2021-Sep-28 | Using summary and patterns style. |
+| 2020-Oct-04 | Initial version in Wiki.    |
+
+## Introduction
+
+When building a digital ecosystem API usability becomes a business priority. Success of your ecosystem depends on APIs that are easy to discover, simple to use, fit to purpose, and consistent across your products.
+
+This document offers guidance that Microsoft Graph API developer teams MUST follow to
+ensure that Microsoft Graph has a consistent and easy to use API surface. A new API design should meet the
+following goals:
+
+\- Developer friendly via consistent naming, patterns, and web standards (HTTP,
+REST, JSON)
+
+\- Work well with SDKs in many programming languages.
+
+\- Sustainable & evolvable via clear API contracts.
+
+The Microsoft Graph guidelines are an extension of the Microsoft REST API
+guidelines. Readers are assumed also be reading and following the Microsoft REST API
+guidelines except where this document outlines specific differences or exceptions to those guidelines.
+Together these guidelines and a library of API patterns serve as the means by
+which API teams discuss and come to consensus on API review recommendations.
+
+This document borrows from multiple public sources such as:
+
+1.  Microsoft Azure REST API Guidelines
+
+2.  Google Cloud Platform APIs
+
+3.  WSO2 Rest API Design Guidelines and others.
+
+Technology and software are constantly changing and evolving, and as such, this
+is intended to be a living document. [Open an
+issue](https://github.com/microsoft/api-guidelines/issues/new/choose) to suggest
+a change or propose a new idea.
+
+### Legend
+
+This document offers prescriptive guidance labeled as follows:
+
+:heavy_check_mark: **MUST** satisfy this specification. If not following this
+advice, you MUST disclose your reason during the Graph API review.
+
+:no_entry: **MUST NOT** use this pattern. If not following this advice, you MUST
+disclose your reason during the Graph API review.
+
+:ballot_box_with_check: **SHOULD** fulfill this specification. If not
+following this advice, you MUST disclose your reason during the Graph API
+review.
+
+:warning: **SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during the Graph API review.
```

Comment:
Don't repeat this "If not following sentence"  Put it at the top or bottom instead.